### PR TITLE
docker: add windows build dockerfiles

### PR DESCRIPTION
**Pull request by gdams**

### CODE REVIEW

**Comment by aahlenst** [comment]
We have Ansible playbooks to provision Windows servers. Why don't we reuse them for the containers?

**Comment by gdams** [comment]
Because you cannot run ansible-playbooks from a windows host. Some people have tried with WSL which is a major hack and adds a lot od bloat to the image.

**Comment by aahlenst** [comment]
Ah, damn. Forgot that you cannot provision Windows containers on an Ansible-compatible box.

File: ansible/Dockerfile.Windows2016_Base [hunk 1]
```diff
@@ -10,25 +10,16 @@ RUN $ProgressPreference = 'SilentlyContinue'; `
...
-    Invoke-WebRequest 'https://mirrors.kernel.org/sourceware/cygwin/x86_64/release/grep/grep-2.27-2.tar.xz' -OutFile C:\cygwin64\tmp\grep.tar.xz; `
-    Start-Process -Wait -FilePath C:\cygwin64\bin\bash.exe -ArgumentList '-l -c "tar -C / -xvf /tmp/grep.tar.xz"'; `
+    Invoke-WebRequest 'https://ci.adoptopenjdk.net/userContent/winansible/cygwin64.tar.xz' -OutFile C:\temp\cygwin64.tar.xz; `
+    Invoke-WebRequest 'https://ci.adoptopenjdk.net/userContent/winansible/cygwin64/bin/*zip*/bin.zip' -OutFile C:\temp\cyg_toolset.zip; `
+    Expand-Archive -LiteralPath C:\temp\cyg_toolset.zip -DestinationPath C:\temp; `
+    C:\temp\bin\tar --force-local -xvf C:\temp\cygwin64.tar.xz C:\ ; `
+    # Remove speech mark from the %PATH%
+    setx /M PATH "%PATH:\"=%" ; `
+    # Add c:\cygwin64\bin to front of %PATH%
+    setx /M PATH "C:\cygwin64\bin;%PATH%" ; `
```
Comment (Willsparker) [inline ansible/Dockerfile.Windows2016_Base:20]: Just to say, I found a different way of doing removing the speech mark with https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1495 as I had a lot of trouble with the `setx` syntax in the context of Ansible. If this works, awesome :)

**Comment by Willsparker** [comment]
```
The string is missing the terminator: ".
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordEx 
   ception
    + FullyQualifiedErrorId : TerminatorExpectedAtEndOfString
```
On the check

**Review by Willsparker — APPROVED** [review]
Looks to be working now! Cool :-)

**Pull request merged**